### PR TITLE
feat: added github auth

### PR DESCRIPTION
**Pull request by pilatmartin**
- Added GitHub OAuth
- Added simple signin/signout UI for testing purposes
- Added middleware which marks all routes starting with `/auth` as protected (require login)
    - `/auth/test` - example protected route
    - redirects to `/` when trying to access anonymously
- Replaced `randomUUID` from `crypto` with `uuid` package due to following error:
image
- Fixed pipeline libsql .env issue
image

**Signed out**
image

**Signed in**
image

**Protected page**
image

### CODE REVIEW

File: quizeek/src/server-actions/user/auth.ts
Comment (DominikBorbuliak) [inline quizeek/src/server-actions/user/auth.ts:1]: I would suggest to call file as `sign-out-action.ts`, If we want to have separate file for each action. But I am open to discussion.

File: quizeek/src/server-actions/user/auth.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use server';
+
+import { signOut } from '@/auth';
+
+export const signOutAction = async (): Promise<void> => {
```
Comment (DominikBorbuliak) [inline quizeek/src/server-actions/user/auth.ts:5]: You did not alter the files in `server-actions/user/index.ts` and `server-actions/index.ts`

**Review by DominikBorbuliak — CHANGES_REQUESTED** [review]

File: quizeek/src/server-actions/user/auth.ts (hunk 1 above)
Comment (Tomas00217) [reply quizeek/src/server-actions/user/auth.ts:5]: That action is not being used at the moment, I think we could get rid of it.

**Review by DominikBorbuliak — APPROVED** [review]
All good!

**Pull request merged**